api/src/main/java/org/openmrs/module/sync2/api/service/impl/SyncRequestWrapperServiceImpl.java

```diff
@@ -0,0 +1,112 @@
+
+@Component("sync2.syncRequestWrapperService")
+public class SyncRequestWrapperServiceImpl implements SyncRequestWrapperService {
+
+    @Autowired
+    private SyncConfigurationService configuration;
+
+    private RestTemplate restTemplate = new RestTemplate();
+
+    @Override
+    public ResponseEntity<String> getObject(RequestWrapper wrapper) {
+        prepareRestTemplate(wrapper.getClientName());
+        try {
+            RequestEntity<String> req = new RequestEntity<>(wrapper.getRequest().getMethod(), wrapper.getRequest().getUrl());
+            return copyResponseWithContentType(restTemplate.exchange(req, String.class));
+        }
+        catch (HttpClientErrorException e) {
+            return new ResponseEntity<>(e.getMessage(), e.getStatusCode());
+        }
+    }
+
+    @Override
+    public ResponseEntity<String> sendObject(RequestWrapper wrapper) {
+        prepareRestTemplate(wrapper.getClientName());
+        try {
+            ClientHelper helper = ClientHelperFactory.createClient(wrapper.getClientName());
+            Object object = helper.convertToObject(wrapper.getRequest().getBody(), wrapper.getClazz());
+
+            RequestEntity req = new RequestEntity<>(object, wrapper.getRequest().getMethod(), wrapper.getRequest().getUrl());
+            ResponseEntity<String> res = restTemplate.exchange(req, String.class);
+
+            return new ResponseEntity<>(res.getBody(), res.getStatusCode());
+        } catch (HttpClientErrorException e) {
+            return new ResponseEntity<>(e.getMessage(), e.getStatusCode());
+        } catch (ClassNotFoundException e) {
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @Override
+    public ResponseEntity<String> deleteObject(RequestWrapper wrapper) {
+        prepareRestTemplate(wrapper.getClientName());
+        try {
+            return restTemplate.exchange(
+                    wrapper.getRequest().getUrl(),
+                    wrapper.getRequest().getMethod(),
+                    new HttpEntity<Object>(wrapper.getRequest().getBody()),
+                    String.class);
+        } catch (HttpClientErrorException e) {
+            return new ResponseEntity<>(e.getMessage(), e.getStatusCode());
+        }
+    }
+
+    @Override
+    public boolean isRequestAuthenticated(RequestWrapper requestWrapper) {
+        User user = Context.getAuthenticatedUser();
+        if (user == null) return false;
+
+        return isInstanceIdValid(requestWrapper.getInstanceId());
+    }
+
+    private boolean isInstanceIdValid(String instanceId) {
+        return isWhitelistDisabled() || isInstanceIdOnWhitelist(instanceId);
+    }
+
+    private boolean isInstanceIdOnWhitelist(String instanceId) {
+        return configuration.getSyncConfiguration().getWhitelist().getInstanceIds().contains(instanceId) &&
+                configuration.getSyncConfiguration().getWhitelist().isEnabled();
```

Comment:
configuration.getSyncConfiguration().getWhitelist().isEnabled() could be !isInstanceDisabled()